skore/src/skore/sklearn/_plot/data/table_report.py

```diff
@@ -0,0 +1,196 @@
+import weakref
+
+from skrub import _dataframe as sbd
+from skrub._reporting._html import to_html
+from skrub._reporting._serve import open_in_browser
+from skrub._reporting._summarize import summarize_dataframe
+
+from skore.sklearn._plot.data import _plotting
+from skore.sklearn._plot.style import StyleDisplayMixin
+from skore.sklearn._plot.utils import HelpDisplayMixin, ReprHTMLMixin
+
+
+def _subsample(df, n_subsample, subsample_strategy, random_state):
+    if n_subsample is None:
+        return df
+    if subsample_strategy == "head":
+        return sbd.head(df, n_subsample)
+    elif subsample_strategy == "random":
+        return sbd.sample(df, n_subsample, seed=random_state)
+    else:
+        raise ValueError(
+            "'subsample_strategy' options are 'head', 'random', "
+            f"got {subsample_strategy}."
+        )
+
+
+def _require_x_col(x_col, kind):
+    if x_col is None:
+        raise ValueError(f"When {kind=!r}, ``x_col`` is mandatory.")
+
+
+def _check_no_args(x_col, y_col, c_col, kind):
+    params = dict(
+        x_col=x_col,
+        y_col=y_col,
+        c_col=c_col,
+    )
+    for k, v in params.items():
+        if v is not None:
+            raise ValueError(f"When {kind=!r}, this function takes no {k!r} argument.")
+
+
+class TableReportDisplay(StyleDisplayMixin, HelpDisplayMixin, ReprHTMLMixin):
+    def __init__(self, summary, df, column_filters=None):
+        self.summary = summary
+        self.column_filters = column_filters
+        self._df = weakref.ref(df)
+
+    @property
+    def df(self):
+        if (_df := self._df()) is None:
+            raise ValueError("The dataset is not accessible by the report anymore.")
+        return _df
+
+    @classmethod
+    def _compute_data_for_display(cls, df, with_plots=True, title=None):
+        summary = summarize_dataframe(
+            df,
+            with_plots=with_plots,
+            title=title,
+        )
+        return cls(summary, df)
+
+    @StyleDisplayMixin.style_plot
```

Comment:
So @Vincent-Maladiere, you can have a look at the `StyleDisplayMixin` class but in short it will allow to add a `set_style` method by taking `whatever_kwargs` parameters. So in short, in `plot`, we could create a `scatter_kwargs` or any other `_kwargs` method which allow to tweak the plots. I did not yet go into details in the `_plotting` module but it might give you some perspective of changes that can be done.